File: python/calculator.py
"""
Simple Calculator
Performs basic arithmetic operations: addition, subtraction,
multiplication, and division.
"""


def add(a, b):
    return a + b


def subtract(a, b):
    return a - b


def multiply(a, b):
    return a * b


def divide(a, b):
    if b == 0:
        return "Error: Division by zero is not allowed."
    return a / b


def main():
    print("=" * 35)
    print("        SIMPLE CALCULATOR")
    print("=" * 35)

    # Get the two numbers from the user
    try:
        num1 = float(input("Enter the first number: "))
        num2 = float(input("Enter the second number: "))
    except ValueError:
        print("Invalid input. Please enter numeric values only.")
        return

    # Show operation menu
    print("\nChoose an operation:")
    print("1. Addition (+)")
    print("2. Subtraction (-)")
    print("3. Multiplication (*)")
    print("4. Division (/)")

    choice = input("Enter choice (1/2/3/4): ").strip()

    if choice == "1":
        result = add(num1, num2)
        symbol = "+"
    elif choice == "2":
        result = subtract(num1, num2)
        symbol = "-"
    elif choice == "3":
        result = multiply(num1, num2)
        symbol = "*"
    elif choice == "4":
        result = divide(num1, num2)
        symbol = "/"
    else:
        print("Invalid choice. Please select 1, 2, 3, or 4.")
        return

    print("\n" + "-" * 35)
    print(f"Result: {num1} {symbol} {num2} = {result}")
    print("-" * 35)


if __name__ == "__main__":
    main()

File: C++/calculator.cpp

#include <iostream>
using namespace std;

int main() {
    double num1, num2, result;
    int choice;

    do {
        cout << "\n===== Simple Calculator =====" << endl;
        cout << "1. Addition (+)" << endl;
        cout << "2. Subtraction (-)" << endl;
        cout << "3. Multiplication (*)" << endl;
        cout << "4. Division (/)" << endl;
        cout << "5. Exit" << endl;
        cout << "Enter your choice (1-5): ";
        cin >> choice;

        if (choice == 5) {
            cout << "Exiting calculator. Goodbye!" << endl;
            break;
        }

        if (choice < 1 || choice > 5) {
            cout << "Invalid choice! Please select between 1 and 5." << endl;
            continue;
        }

        cout << "Enter first number: ";
        cin >> num1;
        cout << "Enter second number: ";
        cin >> num2;

        switch (choice) {
            case 1:
                result = num1 + num2;
                cout << "Result: " << num1 << " + " << num2 << " = " << result << endl;
                break;
            case 2:
                result = num1 - num2;
                cout << "Result: " << num1 << " - " << num2 << " = " << result << endl;
                break;
            case 3:
                result = num1 * num2;
                cout << "Result: " << num1 << " * " << num2 << " = " << result << endl;
                break;
            case 4:
                if (num2 != 0) {
                    result = num1 / num2;
                    cout << "Result: " << num1 << " / " << num2 << " = " << result << endl;
                } else {
                    cout << "Error: Division by zero is not allowed!" << endl;
                }
                break;
        }

    } while (true);

    return 0;
}